dev/com.ibm.ws.kernel.feature.cmdline/src/com/ibm/ws/kernel/feature/internal/generator/FeatureList.java

```diff
@@ -404,13 +403,12 @@ private String toJavaEE(Map<String, Object> version) {
             }
             for (FeatureResource res : fd.getConstituents(SubsystemContentType.FEATURE_TYPE)) {
                 ProvisioningFeatureDefinition otherFD = mfp.getFeatureDefinitions().get(res.getSymbolicName());
-                // This will NPE when we call getJavaVersion, so fail more helpfully
-                if (otherFD == null)
+                // This will NPE when we call getJavaVersion, so skip
```

Comment:
What is the meaning of "skip" here?  Does it mean that the feature will not be considered?  if we aren't able to detect the java version, I'd like to assume we are just on the newest known version of java, which is what other libs like commons-lang3 do now given the new release cadence of java

Reply:
what skip means is we ignore the fact that the feature definition isn't found. The reason this was added was because for compatibility with WebSphere Liberty a load of Open Liberty features define EE 6 feature versions as a baseline, but those features never exist in Open Liberty. As a result the feature list generation just failed. By skipping we will ignore it and move onto the next feature. Including looking at the Java EE 7 one.

For a concrete scenario if you have a feature that depends on EE 7, but tolerates EE 8 if you don't have EE 8 installed it'll list Java SE 8 as being the prereq. If you have EE 7 installed you'll get Java SE 7 & 8.